File: app/src/main/java/com/caramelheaven/lennach/models/network/PostResponse.java
package com.caramelheaven.lennach.models.network;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class PostResponse {
    @SerializedName("banned")
    @Expose
    private Integer bannned;
    @SerializedName("closed")
    @Expose
    private Integer closed;
    @SerializedName("comment")
    @Expose
    private String comment;
    @SerializedName("date")
    @Expose
    private String date;
    @SerializedName("email")
    @Expose
    private String email;
    @SerializedName("endless")
    @Expose
    private Integer endless;
    @SerializedName("files")
    @Expose
    private List<FileResponse> files;
    @SerializedName("files_count")
    @Expose
    private Integer filesCount;
    @SerializedName("lasthit")
    @Expose
    private Integer lasthit;
    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("num")
    @Expose
    private String num;
    @SerializedName("op")
    @Expose
    private Integer op;
    @SerializedName("parent")
    @Expose
    private String parent;
    @SerializedName("posts_count")
    @Expose
    private String postsCount;
    @SerializedName("sticky")
    @Expose
    private Integer sticky;
    @SerializedName("subject")
    @Expose
    private String subject;
    @SerializedName("tags")
    @Expose
    private String tags;
    @SerializedName("timestamp")
    @Expose
    private Integer timestamp;
    @SerializedName("trip")
    @Expose
    private String trip;

    public Integer getBannned() {
        return bannned;
    }

    public Integer getClosed() {
        return closed;
    }

    public String getComment() {
        return comment;
    }

    public String getDate() {
        return date;
    }

    public String getEmail() {
        return email;
    }

    public Integer getEndless() {
        return endless;
    }

    public List<FileResponse> getFiles() {
        return files;
    }

    public Integer getFilesCount() {
        return filesCount;
    }

    public Integer getLasthit() {
        return lasthit;
    }

    public String getName() {
        return name;
    }

    public String getNum() {
        return num;
    }

    public Integer getOp() {
        return op;
    }

    public String getParent() {
        return parent;
    }

    public String getPostsCount() {
        return postsCount;
    }

    public Integer getSticky() {
        return sticky;
    }

    public String getSubject() {
        return subject;
    }

    public String getTags() {
        return tags;
    }

    public Integer getTimestamp() {
        return timestamp;
    }

    public String getTrip() {
        return trip;
    }
}

File: app/src/main/java/com/caramelheaven/lennach/models/network/base/PostResponse.kt
package com.caramelheaven.lennach.models.network.base

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

/**
 * Created by CaramelHeaven on 16:17, 03/02/2019.
 */
class PostResponse {
    @SerializedName("banned")
    @Expose
    val bannned: Int? = null
    @SerializedName("closed")
    @Expose
    val closed: Int? = null
    @SerializedName("comment")
    @Expose
    val comment: String? = null
    @SerializedName("date")
    @Expose
    val date: String? = null
    @SerializedName("email")
    @Expose
    val email: String? = null
    @SerializedName("endless")
    @Expose
    val endless: Int? = null
    @SerializedName("files")
    @Expose
    val files: List<FileResponse>? = null
    @SerializedName("files_count")
    @Expose
    val filesCount: Int? = null
    @SerializedName("lasthit")
    @Expose
    val lasthit: Int? = null
    @SerializedName("name")
    @Expose
    val name: String? = null
    @SerializedName("num")
    @Expose
    val num: String? = null
    @SerializedName("op")
    @Expose
    val op: Int? = null
    @SerializedName("parent")
    @Expose
    val parent: String? = null
    @SerializedName("posts_count")
    @Expose
    val postsCount: String? = null
    @SerializedName("sticky")
    @Expose
    val sticky: Int? = null
    @SerializedName("subject")
    @Expose
    val subject: String? = null
    @SerializedName("tags")
    @Expose
    val tags: String? = null
    @SerializedName("timestamp")
    @Expose
    val timestamp: Int? = null
    @SerializedName("trip")
    @Expose
    val trip: String? = null
}
